Sources/Basics/Archiver+Zip.swift

```diff
@@ -70,6 +71,48 @@ public struct ZipArchiver: Archiver, Cancellable {
         }
     }
 
+    public func compress(
+        directory: AbsolutePath,
+        to destinationPath: AbsolutePath,
+        completion: @escaping (Result<Void, Error>) -> Void
+    ) {
+        do {
+            guard self.fileSystem.isDirectory(directory) else {
+                throw FileSystemError(.notDirectory, directory)
+            }
+
+#if os(Windows)
+            let process = TSCBasic.Process(
+                // FIXME: are these the right arguments?
+                arguments: ["tar.exe", "-a", "-c", "-f", destinationPath.pathString, directory.basename],
```

Comment:
Can tar really produce ZIP archives on Windows? That seems quite surprising to me.

Reply:
https://techcommunity.microsoft.com/t5/containers/tar-and-curl-come-to-windows/ba-p/382409

Reply:
Maybe we should add a `precondition` on `destinationPath` here? `-a` detects the archive format from path extension (and sadly that’s the only option to create a zip archive for `tar.exe`). We may want to make sure we’re actually creating a `zip` archive.

Reply:
Ah that's nice, wish the macOS/Linux tar did the same :)